src/compiler/resolutionCache.ts

```diff
@@ -346,8 +355,28 @@ namespace ts {
             return primaryResult;
         }
 
-        function resolveTypeReferenceDirective(typeReferenceDirectiveName: string, containingFile: string | undefined, options: CompilerOptions, host: ModuleResolutionHost, redirectedReference?: ResolvedProjectReference, _containingSourceFile?: SourceFile, resolutionMode?: SourceFile["impliedNodeFormat"] | undefined): CachedResolvedTypeReferenceDirectiveWithFailedLookupLocations {
-            return ts.resolveTypeReferenceDirective(typeReferenceDirectiveName, containingFile, options, host, redirectedReference, typeReferenceDirectiveResolutionCache, resolutionMode);
+        function resolveTypeReferenceDirective(typeReferenceDirectiveName: string, containingFile: string | undefined, compilerOptions: CompilerOptions, host: ModuleResolutionHost, redirectedReference?: ResolvedProjectReference, _containingSourceFile?: SourceFile, resolutionMode?: SourceFile["impliedNodeFormat"] | undefined): CachedResolvedTypeReferenceDirectiveWithFailedLookupLocations {
+            const primaryResult = ts.resolveTypeReferenceDirective(typeReferenceDirectiveName, containingFile, compilerOptions, host, redirectedReference, typeReferenceDirectiveResolutionCache, resolutionMode);
+            if (!resolutionHost.getGlobalCache || primaryResult.resolvedTypeReferenceDirective) {
+                return primaryResult;
+            }
+            const globalCache = resolutionHost.getGlobalCache();
+            if (globalCache !== undefined && !isExternalModuleNameRelative(typeReferenceDirectiveName)) {
+                const { resolvedTypeReferenceDirective, failedLookupLocations } = loadTypeReferenceDirectiveFromGlobalCache(
+                    Debug.checkDefined(resolutionHost.globalCacheResolutionModuleName)(typeReferenceDirectiveName),

```

Comment:
This change here should also handle node.. so i dont think we need to worry about including typings files at all..

Reply:
So getTypingName to do ATA for unresolved imports + types array ?